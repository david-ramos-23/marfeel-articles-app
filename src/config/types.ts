export interface ThemeProps {
  colors: {
    [color: string]: string;
  };
  sizes: {
    [size: string]: string;
  };
  sideMenu: {
    headerBackground: string;
    bodyBackground: string;
    linkColor: string;
    linkColorActive: string;
    fontFamily: string;
    fontWeight: string;
    fontSize: string;
    items: string[];
  };
  header: {
    background: string;
    fontFamily: string;
  };
}
