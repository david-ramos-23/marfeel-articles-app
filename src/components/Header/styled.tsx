import styled from 'styled-components';

export const StyledHeader = styled.header`
  position: fixed;
  top: 0;
  display: flex;
  flex-direction: column;
  width: 100%;
  background: ${(props) => props.theme.header.background};
  color: ${(props) => props.theme.colors.secondary};
`;

export const StyledHeaderTop = styled.section`
  display: flex;
  flex-wrap: no-wrap;
  align-items: center;
  padding: ${(props) =>
    `${props.theme.sizes.size1} ${props.theme.sizes.size2}`};
`;

export const StyledHeaderTitle = styled.div`
  display: flex;
  flex-grow: 1;
  justify-content: center;
`;
