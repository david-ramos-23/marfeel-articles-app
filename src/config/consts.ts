export const API_URL = 'http://localhost:3001';
export const fonts = [
  { font: 'Roboto', weights: ['100', '100i', '300', '300i', '400', '400i'] },
  { font: 'Open Sans', weights: ['100', '200', '400'] },
];
