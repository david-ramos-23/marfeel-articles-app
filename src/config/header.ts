export const HEADER_WAYPOINTS = {
  firstWaypoint: 400,
  secondWaypont: 800,
};
