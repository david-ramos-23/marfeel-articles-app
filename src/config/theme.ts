import logo from '../../assets/img/logo-marfeel.svg';
import { ThemeProps } from './types';

export const theme: ThemeProps = {
  colors: {
    primary: 'black',
    secondary: 'white',
  },
  sizes: {
    size1: '10px',
    size2: '15px',
    size3: '20px',
    size4: '40px',
    size5: '50px',
  },
  sideMenu: {
    headerBackground: 'rgb(2,0,36)',
    bodyBackground: 'white',
    linkColor: '#b9b9b9',
    linkColorActive: '#de0000',
    fontFamily: 'Roboto',
    fontWeight: '400',
    fontSize: '16px',
    items: [
      'HOME',
      'MAIN SECTION 1',
      'MAIN SECTION 2',
      'MAIN SECTION 3',
      'MAIN SECTION 4',
      'MAIN SECTION 5',
      'MAIN SECTION 6',
      'MAIN SECTION 7',
      'MAIN SECTION 8',
      'MAIN SECTION 9',
    ],
  },
  header: {
    background: 'rgb(0,0,0)',
    fontFamily: 'Roboto',
  },
  footer: {
    background: '#f2f2f2',
    textColor: '#888',
    fontFamily: 'Open Sans',
    fontWeight: '200',
    fontSize: '12px',
  },
};
