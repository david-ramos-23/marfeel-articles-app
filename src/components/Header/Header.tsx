import React from 'react';
import { Logo } from '../shared/Logo';
import { BurgerMenu } from './components/BurgerMenu';
import { Section } from './components/Section';
import { StyledHeader, StyledHeaderTitle, StyledHeaderTop } from './styled';
import { HeaderState } from './types';
import { HEADER_WAYPOINTS } from '../../config/header';

let ticking = false;

export class Header extends React.PureComponent<unknown, HeaderState> {
  public state = {
    isNavigationVisible: window.scrollY <= HEADER_WAYPOINTS.firstWaypoint,
    isHeaderVisible: window.scrollY <= HEADER_WAYPOINTS.secondWaypont,
  };

  private handleScroll = () => {
    const { isNavigationVisible, isHeaderVisible } = this.state;
    const newNavigationState = window.scrollY <= HEADER_WAYPOINTS.firstWaypoint;
    const newHeaderState = window.scrollY <= HEADER_WAYPOINTS.secondWaypont;
    if (
      !ticking &&
      (isHeaderVisible !== newHeaderState ||
        isNavigationVisible !== newNavigationState)
    ) {
      ticking = true;

      this.setState({
        isNavigationVisible: newNavigationState,
        isHeaderVisible: newHeaderState,
      });

      ticking = false;
    }
  };

  public componentDidMount(): void {
    window.addEventListener('scroll', this.handleScroll, true);
  }

  public componentWillUnmount(): void {
    window.removeEventListener('scroll', this.handleScroll, true);
  }

  public render(): JSX.Element {
    const { isHeaderVisible, isNavigationVisible } = this.state;

    if (!isHeaderVisible) {
      return <BurgerMenu showWrapper={true} />;
    }

    return (
      <StyledHeader>
        <StyledHeaderTop>
          <BurgerMenu />
          <StyledHeaderTitle>
            <Logo />
          </StyledHeaderTitle>
        </StyledHeaderTop>
        {isNavigationVisible && <Section />}
      </StyledHeader>
    );
  }
}
