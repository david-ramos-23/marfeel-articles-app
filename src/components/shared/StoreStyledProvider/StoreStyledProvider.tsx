import React from 'react';
import { Provider } from 'react-redux';
import { ThemeProvider } from 'styled-components';

import configureStore from '../../../store';
import { defaultTheme } from '../../../store/theme/theme';
import { SectionProvider } from '../../../hooks/useSection';

export const StoreStyledProvider: React.FunctionComponent = ({ children }) => (
  <Provider store={configureStore()}>
    <ThemeProvider theme={defaultTheme}>
      <SectionProvider>
        <>{children}</>
      </SectionProvider>
    </ThemeProvider>
  </Provider>
);
