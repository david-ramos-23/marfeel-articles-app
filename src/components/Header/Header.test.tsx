import { shallow } from 'enzyme';
import React from 'react';
import renderer from 'react-test-renderer';

import 'jest-styled-components';

import { Header } from './Header';

import { StoreStyledProvider } from '../shared/StoreStyledProvider';
import { HEADER_WAYPOINTS } from '../../config/consts';

describe('Header', () => {
  it('renders without crashing', () => {
    const header = renderer
      .create(
        <StoreStyledProvider>
          <Header />
        </StoreStyledProvider>
      )
      .toJSON();
    expect(header).toMatchSnapshot();
  });

  describe('handleScroll method', () => {
    const wrapper: any = shallow(<Header />);

    it('should be displayed the entire header initially', () => {
      expect(wrapper.state()).toEqual({
        isNavigationVisible: true,
        isHeaderVisible: true,
      });
    });

    it(`it should hide the menu when the user scrolls more than ${HEADER_WAYPOINTS.firstWaypoint}`, () => {
      (global as any).scrollY = HEADER_WAYPOINTS.firstWaypoint + 1;
      wrapper.instance().handleScroll();

      expect(wrapper.state()).toEqual({
        isNavigationVisible: false,
        isHeaderVisible: true,
      });
    });

    it(`it should hide the header when the user scrolls more than ${HEADER_WAYPOINTS.secondWaypont}`, () => {
      (global as any).scrollY = HEADER_WAYPOINTS.secondWaypont + 1;
      wrapper.instance().handleScroll();
      expect(wrapper.state()).toEqual({
        isNavigationVisible: false,
        isHeaderVisible: false,
      });
    });
  });
});
