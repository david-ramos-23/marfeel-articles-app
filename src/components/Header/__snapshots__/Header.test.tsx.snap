// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders without crashing 1`] = `
.c5 {
  width: 100px;
  height: 25px;
}

.c2 {
  padding: 0;
  background: none;
  border: none;
  -webkit-appearance: unset;
  width: 20px;
  height: 16px;
  position: absolute;
  cursor: pointer;
}

.c3 {
  width: 20px;
  height: 2px;
  position: absolute;
  top: 7px;
  border-radius: 2px;
  background-color: white;
}

.c3:before,
.c3:after {
  position: absolute;
  display: block;
  content: '';
  width: 20px;
  height: 2px;
  background-color: white;
  border-radius: 2px;
}

.c3:before {
  top: -6px;
}

.c3:after {
  bottom: -6px;
}

.c6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 20px;
  overflow: hidden;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-family: 'Roboto',sans-serif;
  font-size: 14px;
  font-weight: 400;
}

.c7 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: 50px;
  padding: 0 15px;
  margin: 0;
  list-style: none;
  overflow: auto;
}

.c8 {
  padding-right: 15px;
  color: white;
  white-space: nowrap;
}

.c9 {
  display: block;
  padding-bottom: 2px;
  border-bottom: 2px solid white;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c10 {
  display: block;
  padding-bottom: 2px;
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0 {
  position: fixed;
  top: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: 100%;
  background: rgb(0,0,0);
  color: white;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: no-wrap;
  -ms-flex-wrap: no-wrap;
  flex-wrap: no-wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 10px 15px;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<header
  className="c0"
>
  <div />
  <section
    className="c1"
  >
    <button
      className="c2"
      onClick={[Function]}
    >
      <div
        className="c3"
      />
    </button>
    <div
      className="c4"
    >
      <img
        alt="Logo"
        className="c5"
        src="logo-marfeel.svg"
      />
    </div>
  </section>
  <nav
    className="c6"
  >
    <ul
      className="c7"
    >
      <li
        className="c8"
      >
        <a
          className="c9"
          href="#1"
          onClick={[Function]}
          selected={true}
        >
          HOME
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#2"
          onClick={[Function]}
          selected={false}
        >
          TRAVELS
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#3"
          onClick={[Function]}
          selected={false}
        >
          TECH
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#4"
          onClick={[Function]}
          selected={false}
        >
          NATIONAL
        </a>
      </li>
      <li
        className="c8"
      >
        <a
          className="c10"
          href="#5"
          onClick={[Function]}
          selected={false}
        >
          INTERNATIONAL
        </a>
      </li>
    </ul>
  </nav>
</header>
`;
